src/System.Binary.Base64/System/Binary/BufferExtensions.cs

```diff
@@ -0,0 +1,116 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Buffers;
+using System.Collections.Sequences;
+
+// This should most likely be in System.Text.Primitives and not here. Will move once Base64 moves.
+namespace System.Binary.Base64
+{
+    public static class BufferExtensions
+    {
+        const int stackLength = 32;
+
+        public static void Pipe(this ITransformation transformation, ReadOnlyBytes source, IOutput destination)
+        {
+            int afterMergeSlice = 0;
+            ReadOnlySpan<byte> remainder;
+            Span<byte> stackSpan;
+
+            unsafe
+            {
+                byte* stackBytes = stackalloc byte[stackLength];
+                stackSpan = new Span<byte>(stackBytes, stackLength);
+            }
+
+            var poisition = Position.First;
+            while (source.TryGet(ref poisition, out var sourceBuffer, true))
```

Comment:
You might want to check if it's not worth optimizing this for a single segment ROB. i.e. if ROB has one segment, there is no need to TryGet and instead just access the First property.